colcon_core/package_descriptor.py

```diff
@@ -87,17 +88,18 @@ def get_recursive_dependencies(
           categories
         :param Iterable[str] recursive_categories: The names of the recursive
           categories
-        :returns: The names of the recursive dependencies
+        :returns: DependencyDescriptor objects of recursive dependencies
```

Comment:
Please mention the type of the return value as part of the `rtype`:

```
:returns: The recursive dependencies
:rtype: set[DependencyDescriptor]
```

Reply:
Updated both docstrings.